quickstarts/microsoft.web/function-app-storage-private-endpoints/main.bicep

```diff
@@ -0,0 +1,442 @@
+@description('The location into which the resources should be deployed.')
+param location string = resourceGroup().location
+
+@description('The name of the Azure Function app.')
+param functionAppName string = 'func-${uniqueString(resourceGroup().id)}'
+
+@description('The name of the Azure Function hosting plan.')
+param functionAppPlanName string = 'plan-${uniqueString(resourceGroup().id)}'
+
+@description('Determines if a Windows or Linux-based hosting plan is created.  Set to true for Linux, and false for Windows.')
+param isReserved bool = false
```

Comment:
The "reserved" term comes from the Microsoft.Web/serverFarm resource - https://docs.microsoft.com/en-us/azure/templates/microsoft.web/serverfarms?tabs=json#appserviceplanproperties-object. I'll change the parameter description to more closely match what's on the documentation for the "reserved" property.

Reply:
Got it - for the user, is the user expected to pick Linux vs. Windows?  Or reserved vs. not?  My guess it user doesn't care about the implementation of the web app, just whether they want linux v. windows.  And if that's accurate, then we should probably rename the parameter accordingly...

Reply:
The term "reserved" is a poor name to indicate if the function app is based on Windows or Linux.  If reserved, that means use Linux, otherwise use Windows.

Do you think reworking to something like "functionPlanOS" with allowed values of "Windows" and "Linux" would be easier?  If so, I could probably rework some of the conditional logic in the .bicep file to change the necessary values if the value is Windows or Linux.

Reply:
yeah I think your logic is more or less the same regardless of the name... but having user choose Windows/Linux seems to be the right pivot.  functionPlanOS seems fine